cpp/src/arrow/util/future.h

```diff
@@ -493,6 +493,22 @@ class ARROW_MUST_USE_TYPE Future {
     });
   }
 
+  /// \brief Implicit constructor to create a future from a Result, enabling use
+  ///     of macros like ARROW_ASSIGN_OR_RAISE.
+  Future(Result<ValueType> res) : Future() {  // NOLINT(runtime/explicit)
```

Comment:
Some tools don't recognize the parenthesized form
```suggestion
  Future(Result<ValueType> res) : Future() {  // NOLINT runtime/explicit
```